packages/network-controller/src/NetworkController.ts

```diff
@@ -2438,6 +2443,13 @@ export class NetworkController extends BaseController<
     }
 
     if (mode === 'add' || mode === 'update') {
+      if (
+        JSON.stringify(
+          state.networkConfigurationsByChainId[args.networkFields.chainId],
+        ) !== JSON.stringify(args.networkConfigurationToPersist)
+      ) {
+        args.networkConfigurationToPersist.lastUpdatedAt = Date.now();
+      }
```

Comment:
Some tests (see below) state that the state doesn't change, yet they technically do run the state change method here.

To ensure test consistency, I've added this additional branch to check the stringified contents before and after to see if network changed.


https://github.com/MetaMask/core/blob/f47d5b47c5f33607fa4cb277987c2a2699d86d26/packages/network-controller/tests/NetworkController.test.ts#L10951-L10952

Reply:
LMK if devs know of a nicer way of checking network configuration equality. It's a little tricky since it is a complex object.

Reply:
Gotcha. What are your thoughts on using a package like `fast-deep-equal` to perform a deep equality check?

Reply:
Oooh dope! If this is a package we have installed I'm down to use it!

Reply:
Added fast-deep-equal. Great callout, some of our other libs use it too!